app/upgrades/v12/upgrades.go

```diff
@@ -72,26 +72,44 @@ func CreateV12UpgradeHandler(
 			HostEnabled: true,
 			AllowMessages: []string{
 				sdk.MsgTypeURL(&banktypes.MsgSend{}),
+				sdk.MsgTypeURL(&banktypes.MsgMultiSend{}),
 				sdk.MsgTypeURL(&stakingtypes.MsgDelegate{}),
 				sdk.MsgTypeURL(&stakingtypes.MsgBeginRedelegate{}),
+				sdk.MsgTypeURL(&stakingtypes.MsgUndelegate{}),
 				sdk.MsgTypeURL(&stakingtypes.MsgCreateValidator{}),
 				sdk.MsgTypeURL(&stakingtypes.MsgEditValidator{}),
 				sdk.MsgTypeURL(&distrtypes.MsgWithdrawDelegatorReward{}),
 				sdk.MsgTypeURL(&distrtypes.MsgSetWithdrawAddress{}),
 				sdk.MsgTypeURL(&distrtypes.MsgWithdrawValidatorCommission{}),
 				sdk.MsgTypeURL(&distrtypes.MsgFundCommunityPool{}),
 				sdk.MsgTypeURL(&govtypes.MsgVote{}),
+				sdk.MsgTypeURL(&govtypes.MsgVoteWeighted{}),
+				sdk.MsgTypeURL(&govtypes.MsgSubmitProposal{}),
+				sdk.MsgTypeURL(&govtypes.MsgDeposit{}),
 				sdk.MsgTypeURL(&authz.MsgExec{}),
 				sdk.MsgTypeURL(&authz.MsgGrant{}),
 				sdk.MsgTypeURL(&authz.MsgRevoke{}),
-				// wasm msgs here
+				// wasm
 				sdk.MsgTypeURL(&wasmtypes.MsgStoreCode{}),
 				sdk.MsgTypeURL(&wasmtypes.MsgInstantiateContract{}),
 				sdk.MsgTypeURL(&wasmtypes.MsgInstantiateContract2{}),
 				sdk.MsgTypeURL(&wasmtypes.MsgExecuteContract{}),
 				sdk.MsgTypeURL(&wasmtypes.MsgMigrateContract{}),
 				sdk.MsgTypeURL(&wasmtypes.MsgUpdateAdmin{}),
 				sdk.MsgTypeURL(&wasmtypes.MsgClearAdmin{}),
+				sdk.MsgTypeURL(&wasmtypes.MsgIBCSend{}),
+				sdk.MsgTypeURL(&wasmtypes.MsgIBCCloseChannel{}),
```

Comment:
These messages should not be included, are not being used in Msg Service. I think it is not used at all.

there is also a special wildcard to accept all messages: `AllowMessages: []string{"*"}`
https://github.com/cosmos/ibc-go/blob/v4.2.0/docs/apps/interchain-accounts/parameters.md#allowmessages

Reply:
Thanks, this is much nicer. In cleanup PR